### PR TITLE
CLN: parametrize test_nat_comparisons

### PR DESCRIPTION
**Pull request by ivanovmg**
Parametrize ``TestDatetime64SeriesComparison.test_nat_comparisons``
using ``operator`` module.

If this kind of cleanup is considered reasonable,
then I will go through the remaining tests in this test module.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
yes this type of cleanup is good. targeted PRs are the best.

cc @jbrockmendel if comments.

**Comment by jbrockmendel** [comment]
i do not find this clearer than the status quo.

the lowest-hanging fruit in the tests/arithmetic directory are ones that do not use the box_with_array fixture

there are also very likely redundant tests that can be de-duplicated

**Comment by jreback** [comment]
@jbrockmendel ok so you are -1 on this?

**Comment by jbrockmendel** [comment]
-0.5, i.e. i dont think this is an improvement, but not a hill to die on if others have strong opinions

**Comment by jreback** [comment]
ok closing this

**Comment by simonjayhawkins** [comment]
> -0.5, i.e. i dont think this is an improvement, but not a hill to die on if others have strong opinions

Parameterisation is beneficial when there a multiple test failures. It is sometimes easier to debug when all the passing and failing test cases are visible together.

This PR looked good to me.

**Comment by jreback** [comment]
ok @ivanovmg can u merge master

**Comment by ivanovmg** [comment]
Merged master.
Also I renamed local variable ``operator`` into ``op`` for consistency with other test and not to confuse with module ``operator``.
However, probably more readable option would be ``operator_`` with a trailing underscore, so it is more clear that we are talking about the operator.
Alternatively, ``binary_operator`` - longer name, but more clear IMHO.

**Comment by jbrockmendel** [comment]
> Parameterisation is beneficial when there a multiple test failures. It is sometimes easier to debug when all the passing and failing test cases are visible together.

This is true in a general sense, but I'd point out:

1) we have many tests where we don't parametrize these particular 6 methods

2) The 6 Series objects get created at test collection time, which increases the memory footprint of the test suite.  In isolation these 6 small Series is not a big deal, but we are already at a point where memory is causing the py38 windows build to fail regularly.

Again, not a hill worth dying on, just saying.

File: pandas/tests/arithmetic/test_datetime64.py [hunk 1]
```diff
@@ -152,8 +152,6 @@ def test_dt64arr_nat_comparison(self, tz_naive_fixture, box_with_array):
 
 
 class TestDatetime64SeriesComparison:
-    # TODO: moved from tests.series.test_operators; needs cleanup
```
Comment (jbrockmendel) [inline pandas/tests/arithmetic/test_datetime64.py:155]: this comment pertains to the entire class, not just this test

Comment (ivanovmg) [reply pandas/tests/arithmetic/test_datetime64.py]: Got it. Moved it back.

**Comment by jreback** [comment]
thanks @ivanovmg

**Pull request merged**